src/main/java/com/atlassian/db/replica/internal/circuitbreaker/BreakOnNotSupportedOperations.java

```diff
@@ -1,30 +1,42 @@
 package com.atlassian.db.replica.internal.circuitbreaker;
 
-import com.atlassian.db.replica.api.circuitbreaker.BreakerState;
-import com.atlassian.db.replica.spi.circuitbreaker.CircuitBreaker;
+import com.atlassian.db.replica.api.SqlCall;
 import com.atlassian.db.replica.internal.ReadReplicaUnsupportedOperationException;
+import com.atlassian.db.replica.spi.circuitbreaker.CircuitBreaker;
 
+import java.sql.SQLException;
 import java.sql.SQLFeatureNotSupportedException;
 
-import static com.atlassian.db.replica.api.circuitbreaker.BreakerState.CLOSED;
-import static com.atlassian.db.replica.api.circuitbreaker.BreakerState.OPEN;
-
 public class BreakOnNotSupportedOperations implements CircuitBreaker {
-    private static volatile BreakerState state = CLOSED;
+
+    private static volatile boolean ALL_CALLS_SUPPORTED_SO_FAR = true;
```

Comment:
Nit. We can re-use cache here. This way, it could be easily extended to use external cache and work across nodes.

Reply:
Oooh, good point

Reply:
Done